Sources/MapboxNavigation/CarPlayManager.swift

```diff
@@ -618,17 +613,35 @@ extension CarPlayManager {
      previewed.
      */
     public func previewRoutes(for indexedRouteResponse: IndexedRouteResponse) {
+        guard shouldPreviewRoutes(for: indexedRouteResponse) else { return }
         let trip = CPTrip(indexedRouteResponse: indexedRouteResponse)
         previewRoutes(for: trip)
     }
     
+    /**
+     Allows to cancel routes preview on CarPlay .
+     */
+    public func cancelRoutesPreview() {
+        guard self.indexedRouteResponse != nil else { return }
+        self.indexedRouteResponse = nil
+        mainMapTemplate?.hideTripPreviews()
+        popToRootTemplate(interfaceController: interfaceController, animated: true)
+        delegate?.carPlayManagerCanceledRoutesPreview(self)
+    }
+    
+    func shouldPreviewRoutes(for indexedRouteResponse: IndexedRouteResponse) -> Bool {
+        guard self.indexedRouteResponse?.currentRoute == indexedRouteResponse.currentRoute else { return true }
+        return self.indexedRouteResponse?.routeResponse.routes != indexedRouteResponse.routeResponse.routes
+    }
+    
     func previewRoutes(for trip: CPTrip) {
         guard let traitCollection = (self.carWindow?.rootViewController as? CarPlayMapViewController)?.traitCollection,
               let interfaceController = interfaceController else {
                   return
               }
         
         let modifiedTrip = delegate?.carPlayManager(self, willPreview: trip) ?? trip
+        self.indexedRouteResponse = modifiedTrip.routeChoices.first?.indexedRouteResponse
```

Comment:
Is this value setting for `indexedRouteResponse` is really needed? 

Because `CarPlayManager.mapTemplate(_:selectedPreviewFor:using:)` is called right after  and it resets the `indexedRouteResponse` value

Reply:
Yes, that's correct. Thanks for the suggestion, I'll update this part.